src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -413,6 +425,10 @@ ov::hetero::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model
         device_config[ov::cache_dir.name()] = "";
         // set exclusive_async_requests in case when model is split
         if (orderedSubgraphs.size() > 1) {
+            m_compiled_submodels[id].model->validate_nodes_and_infer_types();
+            if (m_compiled_submodels[id].model->is_dynamic()) {
+                OPENVINO_THROW("Heterogeneous mode is not supported for dynamic models yet!");
```

Comment:
why not? what prevents us from enabling? 
as far as I remember, we worked on dynamism support and even have (had) tests on it.

Reply:
> why not? what prevents us from enabling? as far as I remember, we worked on dynamism support and even have (had) tests on it.

I'm not aware of this work. Anyway currently it is not working.

Reply:
@ilya-lavrenov I removed this check to avoid regression. All plugin errors are still visible. But anyway additional work will be required to make inference working in dynamism.

Reply:
https://github.com/openvinotoolkit/openvino/pull/8050